### PR TITLE
Add tests

### LINKED ISSUE #53: Add tests
Once the basic functionality has been sorted out, we should split the code into smaller modules and add some tests.

### PR DESCRIPTION
**Pull request by nickmccurdy**
Fixes #53

- [x] Expose code for testing
- [x] Test basic button injection
- [ ] Create fully-functioning stubs
  - [ ] `chrome.extension.storage.sync` (needs to be custom)
  - [ ] `MutationObserver`
- [ ] Attempt to test with mocked `window.location`
- [ ] Test option syncing behavior
- [ ] Scope ESLint config for tests
- [ ] Consider adding more granular unit tests per class/file/method/function
- [ ] Complete test coverage

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,5 +56,17 @@
     "hooks": {
       "pre-commit": "lint-staged"
     }
+  },
+  "babel": {
```
Comment (kaicataldo) [inline package.json]: Can we add this to a babelrc file for consistency?

Comment (nickmccurdy) [reply package.json]: This has the same purpose as `.babelrc` (it's shared between both webpack and jest), but I can extract the config anyway if you prefer.

Comment (kaicataldo) [reply package.json]: We're using .*rc files for our other tooling, and I think it's best to be consistent when possible.

**Comment by kaicataldo** [comment]
This is great! I would advocate for getting the testing infrastructure set up in this PR and landing it so that it frees other people up to start writing tests. What do you think?

**Comment by nickmccurdy** [comment]
I have a bunch of different integration tests for content and options in progress though. What would you consider the testing infrastructure? Just setting up the Jest script and config without any additional libraries?

**Comment by nickmccurdy** [comment]
This got a bit messy with some unrelated test code and configs being combined, so I'm going to split it up into multiple PRs.

**Pull request closed without merging**